### PR TITLE
python3.pkgs.django-pwa: init at 1.1.0

### PR DESCRIPTION
**Pull request by DerDennisOP**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
This should be one commit

File: pkgs/development/python-modules/django-pwa/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  checkPhase = ''
+    runHook preCheck
+    ${python.interpreter} runtests.py
+    runHook postCheck
+  '';
+
+  meta = with lib; {
+    description = "A Django app to include a manifest.json and Service Worker instance to enable progressive web app behavoir";
+    homepage = "https://github.com/jazzband/django-pwa/";
```
Comment (natsukium) [inline pkgs/development/python-modules/django-pwa/default.nix:42]:
```suggestion
    homepage = "https://github.com/silviolleite/django-pwa";
```
This is a dead link.

File: pkgs/development/python-modules/django-pwa/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    changelog = "https://github.com/jazzband/django-pwa/releases/tag/v${version}";
```
Comment (natsukium) [inline pkgs/development/python-modules/django-pwa/default.nix:43]: ditto

File: pkgs/development/python-modules/django-pwa/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  src = fetchFromGitHub {
+    owner = "silviolleite";
+    repo = "django-pwa";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-tP1+Jm9hdvN/ZliuVHN8tqy24/tOK1LUUiJv1xUqRrY=";
+  };
+
+  nativeBuildInputs = [
+    setuptools-scm
```
Comment (natsukium) [inline pkgs/development/python-modules/django-pwa/default.nix:27]:
```suggestion
    setuptools
```
Can we use `setuptools`?

File: pkgs/development/python-modules/django-pwa/default.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+, pytest-django
+, python
+, pythonOlder
+, setuptools-scm
+}:
+
+buildPythonPackage rec {
+  pname = "django-pwa";
+  version = "1.1.0";
+  format = "setuptools";
```
Comment (natsukium) [inline pkgs/development/python-modules/django-pwa/default.nix:15]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/django-pwa/default.nix [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  };
+
+  nativeBuildInputs = [
+    setuptools-scm
+  ];
+
+  propagatedBuildInputs = [
+    django
+  ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/django-pwa/default.nix:33]: Could you please add pythonImportsCheck?

File: pkgs/development/python-modules/django-pwa/default.nix [hunk 6]
```diff
@@ -0,0 +1,47 @@
+{ lib
+, buildPythonPackage
+, django
+, fetchFromGitHub
+, pytest
+, pytest-django
```
Comment (natsukium) [inline pkgs/development/python-modules/django-pwa/default.nix:6]:
```suggestion
```
Please remove unused inputs.

**Comment by natsukium** [comment]
@ofborg build python311Packages.django-pwa

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 235422` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.django-pwa</li>
    <li>python310Packages.django-pwa.dist</li>
    <li>python311Packages.django-pwa</li>
    <li>python311Packages.django-pwa.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**